chainercv/evaluations/eval_semantic_segmentation_iou.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import division
+
+import numpy as np
+import six
+
+
+def calc_semantic_segmentation_confusion(pred_labels, gt_labels):
+    """Collect a confusion matrix.
+
+    The number of classes :math:`n\_class` is computed as the maximum
+    class id among :obj:`pred_labels` and :obj:`gt_labels`.
+
+    Args:
+        pred_labels (iterable of numpy.ndarray): A collection of predicted
+            labels. The shape of a label array
+            is :math:`(H, W)`. :math:`H` and :math:`W`
+            are height and width of the label.
+        gt_labels (iterable of numpy.ndarray): A collection of ground
+            truth label. The shape of a ground truth label array is
+            :math:`(H, W)`. The corresponding prediction label should
+            have the same shape.
+            A pixel with value :obj:`-1` will be ignored during evaluation.
+
+    Returns:
+        numpy.ndarray:
+        A confusion matrix. Its shape is :math:`(n\_class, n\_class)`.
+        The :math:`(i, j)` th element corresponds to the number of pixels
+        that are labeled as class :math:`i` by the ground truth and
+        class :math:`j` by the prediction.
+
+    """
+    pred_labels = iter(pred_labels)
+    gt_labels = iter(gt_labels)
+
+    n_class = 0
+    confusion = np.zeros((n_class, n_class), dtype=np.int64)
+    for pred_label, gt_label in six.moves.zip(pred_labels, gt_labels):
+        if pred_label.ndim != 2 or gt_label.ndim != 2:
+            raise ValueError('ndim of inputs should be two.')
+        if pred_label.shape != gt_label.shape:
+            raise ValueError('Shapes of inputs should be same.')
+        pred_label = pred_label.flatten()
+        gt_label = gt_label.flatten()
+
+        # Dynamically expand the confusion matrix if necessary.
+        lb_max = np.max((pred_label, gt_label))
+        if lb_max >= n_class:
+            expanded_confusion = np.zeros
+            expanded_confusion = np.zeros((lb_max + 1, lb_max + 1))
```

Comment:
Perhaps it is better to use `dtype=np.int64` option. You are using this option at L.36 but it will be overwritten by expansion.